homeassistant/components/recorder/statistics.py

```diff
@@ -550,25 +550,26 @@ def compile_statistics(instance: Recorder, start: datetime) -> bool:
 
     _LOGGER.debug("Compiling statistics for %s-%s", start, end)
     platform_stats: list[StatisticResult] = []
+    old_metadata_dict: dict[str, tuple[int, StatisticMetaData]] = {}
```

Comment:
I know you copied the name from below but I feel like we could pick a better name. Maybe `current_metadata` ?

Reply:
3ef5d4798312cc2e528003ea54773001d8b1d61b